File: model/src/main/java/pl/agh/harmonytools/model/MajorScale.java
package pl.agh.harmonytools.model;

import lombok.Getter;

@Getter
public class MajorScale extends Scale {

    private final int[] pitches = {0, 2, 4, 5, 7, 9, 11};
    private Key key;

    public MajorScale(Key key) {
        super(key);
    }

    public MajorScale(BaseNote basenote, int tonicPitch){
        super(basenote, tonicPitch);
    }
}

File: model/src/main/scala/pl/agh/harmonytools/model/MajorScale.scala
package pl.agh.harmonytools.model

case class MajorScale(private val majorKey: Key) extends Scale {
  override protected val pitches: List[Int] = List(0, 2, 4, 5, 8, 10, 11)
  override protected val key: Key = majorKey
}

object MajorScale {
  def apply(baseNote: BaseNote.BaseNoteType, tonicPitch: Int): MajorScale =
    MajorScale(Key(baseNote, tonicPitch))
}

